### PR TITLE
A canister that transfers funds using ICP Ledger API

### PR DESCRIPTION
**Pull request by roman-kashitsyn**
This change adds an example of a canister that can rewards its most
active users by transferring them some ICP.

This change is still work in progress:
- [x] Code example for transferring funds and handling errors.
- [ ] Instructions how to build and install the ICP Ledger (there is no public ledger with this interface yet).
- [ ] Instructions how to transfer funds using the local ledger.

### CODE REVIEW

File: motoko/ledger-transfer/src/ledger_transfer/Account.mo [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+import CRC32     "./CRC32";
+import SHA224    "./SHA224";
+
+module {
+  // 28-byte array.
+  public type AccountIdentifier = Blob.Blob;
+  // 32-byte array.
+  public type Subaccount = Blob.Blob;
+  // 32-byte array.
+  public type Address = Blob.Blob;
```
Comment (rossberg) [inline motoko/ledger-transfer/src/ledger_transfer/Account.mo:16]:
```suggestion
  public type AccountIdentifier = Blob;
  // 32-byte array.
  public type Subaccount = Blob;
  // 32-byte array.
  public type Address = Blob;
```

**Review by rossberg — COMMENTED** [review]
Very nice!

**Comment by krpeacock** [comment]
@roman-kashitsyn let's set aside the local ledger consideration for now and get the existing example merged

**Comment by nicolas-martin** [comment]
I'm getting an error building the ledger canister. 
```
➜  ledger-transfer git:(roman-ledger-transfer) dfx build ledger
Building canisters...
The post-build step failed for canister 'ryjl3-tyaaa-aaaaa-aaaba-cai' with an embedded error: No such file or directory (os error 2)
```

File: motoko/ledger-transfer/ledger.did [hunk 2]
```diff
@@ -0,0 +1,87 @@
+// This is the official Ledger interface that is guaranteed to be backward compatible.
+
+// Amount of ICP tokens, measured in 10^-8 of a token.
+type ICP = record {
```
Comment (MarioDfinity) [inline motoko/ledger-transfer/ledger.did:4]: Do we want to rename these too to Tokens?

**Review by MarioDfinity — APPROVED** [review]

**Review by krpeacock — APPROVED** [review]
These are helpful examples, thanks!

**Comment by roman-kashitsyn** [comment]
> I'm getting an error building the ledger canister, is it missing `ledger.wasm`?

@nicolas-martin, I described how to setup a local instance of ledger in this document: https://github.com/dfinity/ic/tree/master/rs/rosetta-api/ledger_canister#deploying-locally
Hope that helps

**Pull request merged**